api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/aup/AcceptableUsagePolicyProperties.java

```diff
@@ -80,6 +80,30 @@ public MongoDb() {
          * The table name in the database that holds the AUP attribute to update for the user.
          */
         private String tableName;
+        
+        /**
+         * The column to store the AUP attribute. May differ from the profile attribute defined by <code>aupAttributeName</code>.
+         * SQL query can be further customized by setting <code>sqlUpdateAUP</code>.
+         */
+        private String aupColumn;
+        
+        /**
+         * The column to idetify the principal.
+         * SQL query can be further customized by setting <code>sqlUpdateAUP</code>.
+         */
+        private String principalIdColumn = "username";
+        
+        /**
+         * The profile attribute to extract the value for the <code>principalIdColumn</code> used in the WHERE clause
+         * of <code>sqlUpdateAUP</code>. If empty, the principal ID will be used.
+         */
+        private String principalIdAttribute;
+        
+        /**
+         * The query template to update the AUP attribute.
+         * %s placeholders represent tableName, aupColumn, principalIdColumn settings.
+         */
+        private String sqlUpdateAUP = "UPDATE %s SET %s=true WHERE %s=?";
```

Comment:
This seems rather strange; if one has already defined a principal-id column and an AUP column, then one shouldn't have to define them again in the query. It seems like you can remove the setting, and add one for table name, and then build the query on the fly.

...unless you have use cases where the query is entirely different or complicated and you need full control over it?

Reply:
Yes, the behaviour is exactly as you described. this setting is sort of a last resort for special situations. The use case I had in mind for that setting is normalising the username value to upper/lower case, if required. So, I could easily change the query to `UPDATE %s SET %s=true WHERE lower(%s)=lower(?)`. I came across this when setting up PM Password Resets, where it was essential to be able to change the update query like this. 
If you think this scenario is too odd, I could replace this by a setting for username transformation (NONE,UPPER, LOWER) to adjust the query. From my point of view, this limits the configuration flexibility and may cause issues with some SQL dialects, not support whatever change is made to the query.

Reply:
The field needs to be renamed. This is probably the last change before the merge.